README.md

```diff
@@ -1,2 +1,92 @@
+
+Usually, operators use `ServiceBinding` resource name (`.metadata.name`) as the bindings directory name, but the spec also provides a way to override that name through the `.spec.name` field. So, there is a chance for bindings name collision.  However, due to the nature of the volume mount in Kubernetes, the bindings directory will contain values from only one of the Secret resources.  This is a caveat of using the bindings directory name to look up the bindings.
+
+### Purpose of the type and the provider fields in the workload projection
+
+The specification mandates the `type` field and recommends `provider` field in the projected binding.  In many cases the `type` field should be good enough to select the appropriate binding.  In cases where it is not (e.g. when there are different providers for the same Provisioned Service), the `provider` field may be used.  For example, when the type is `mysql`, the `provider` value might be `mariadb`, `oracle`, `bitnami`, `aws-rds`, etc.  When the workload is selecting a binding, if necessary, it could consider `type` and `provider` as a composite key to avoid ambiguity.  This could be helpful if a workload needs to choose a particular provider based on the deployment environment.  In the deployment environment (`stage`, `prod`, etc.), at any given time, you need to ensure only one binding projection exists for a given `type` or `type` and `provider` -- unless your workload needs to connect to all the services.
+
+### Programming language specific library APIs
+
+A workload can retrieve bindings through a library available for your programming language of choice.  Language-specific APIs are encouraged to follow the pattern described here, but may not.  Consult your library API documentation to confirm its behavior.
+
+For languages such as Go without operator overloading, separate functions can be used to retrieve bindings:
+
+```
+Bindings(_type string) []Binding
+BindingsWithProvider(_type, provider string) []Binding
+```
+
+For languages such as Java with operator overloading, the same method name with different argument lists can be used to retrieve bindings:
+
+```
+public List<Binding> filterBindings(@Nullable String type)
+public List<Binding> filterBindings(@Nullable String type, @Nullable String provider)
+```
+
+(Example taken from [Spring Cloud Bindings](https://github.com/spring-cloud/spring-cloud-bindings))
+
+The specification does not guarantee a single binding of a given type or type & provider tuple so the APIs return collections of bindings.  Depending on your workload need, you can choose to connect to the first entry or all of them.
```

Comment:
can we link out to specific libraries instead? The function signatures here don't help me know what I actually need to do as a workload developer.

Reply:
Now there are links to both Go and Java libraries.